### PR TITLE
gh-105052:update timeit function's description

### PR DESCRIPTION
**Pull request by Cherrymelon**
Update  time unit description in `timeit` function and add note to timeit doc page

* Issue: gh-105052

### CODE REVIEW

File: Lib/timeit.py [hunk 1]
```diff
@@ -77,9 +77,11 @@ def inner(_it, _timer{init}):
     return _t1 - _t0
 """
 
+
 def reindent(src, indent):
     """Helper to reindent a multi-line statement."""
-    return src.replace("\n", "\n" + " "*indent)
+    return src.replace("\n", "\n" + " " * indent)
```
Comment (terryjreedy) [inline Lib/timeit.py:83]: Revert.  See PEP 8, Other Recommendations
```suggestion
    return src.replace("\n", "\n" + " "*indent)
```

File: Lib/timeit.py [hunk 2]
```diff
@@ -228,16 +231,19 @@ def autorange(self, callback=None):
                     return (number, time_taken)
             i *= 10
 
+
```
Comment (terryjreedy) [inline Lib/timeit.py:233]: Methods are single-spaced.  See PEP 8, Blank Lines
```suggestion
```

File: Doc/library/timeit.rst [hunk 3]
```diff
@@ -124,7 +126,7 @@ The module defines three convenience functions and a public class:
 
       Time *number* executions of the main statement.  This executes the setup
       statement once, and then returns the time it takes to execute the main
-      statement a number of times, measured in seconds as a float.
+      statement a number of times, measured in seconds as a float by default timer.
```
Comment (terryjreedy) [inline Doc/library/timeit.rst]:
```suggestion
      statement a number of times.  The default timer returns seconds as a float.
```

Or see other suggestions on issue.

File: Misc/NEWS.d/next/Documentation/2023-05-29-14-10-24.gh-issue-105052.MGFwbm.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+timeit doc update
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Documentation/2023-05-29-14-10-24.gh-issue-105052.MGFwbm.rst]:
```suggestion
Update ``timeit`` doc to specify that time in seconds is just the default.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
The PR could have waited for more discussion on the issue.  See [this comment](https://github.com/python/cpython/issues/105052#issuecomment-1567324058) for what I think the most important change for this issue should be, and where.

The Timer docstring has an obsolete and now wrong statement about the timer that should be deleted: `; the timer function is platform-dependent (see the module doc string)`

File: Lib/timeit.py (hunk 2 above)
Comment (Cherrymelon) [reply Lib/timeit.py:233]: sry,I'm confused about the max blank number defination around top-level class and function.It shouldn't be 2 blank?If I changed to one blank,IDE tell me `PEP 8: E302 expected 2 blank lines, found 1`

File: Lib/timeit.py (hunk 1 above)
Comment (Cherrymelon) [reply Lib/timeit.py:83]: Pycharm default formatter add spaces ignore the priority of statements.Any good advice?

File: Lib/timeit.py (hunk 2 above)
Comment (terryjreedy) [reply Lib/timeit.py:233]: Ignore.

File: Lib/timeit.py [hunk 5]
```diff
@@ -166,7 +168,8 @@ def timeit(self, number=default_number):
 
         To be precise, this executes the setup statement once, and
         then returns the time it takes to execute the main statement
-        a number of times, as a float measured in seconds.  The
+        a number of times, as a float measured in seconds if using default
+        timer.Thus,time unit could be changed if giving other timer.  The
```
Comment (terryjreedy) [inline Lib/timeit.py]:
```suggestion
        a number of times, as a float measured in seconds if using the default
        timer. Thus, the time unit could be changed if using other timer.  The
```

**Review by terryjreedy — APPROVED** [review]

File: Lib/timeit.py [hunk 6]
```diff
@@ -166,7 +168,8 @@ def timeit(self, number=default_number):
 
         To be precise, this executes the setup statement once, and
         then returns the time it takes to execute the main statement
-        a number of times, as a float measured in seconds.  The
+        a number of times, as a float measured in seconds if using the default
+        timer. Thus, the time unit could be changed if using other timer.  The
```
Comment (terryjreedy) [inline Lib/timeit.py]:
```suggestion
        a number of times, as float seconds if using the default timer.   The
```

**Pull request merged**